sport-moment-nft/contracts/EnglishPremierLeague.cdc

```diff
@@ -0,0 +1,843 @@
+    Author: Jeremy Ahrens jer.ahrens@dapperlabs.com
+*/
+
+import FungibleToken from 0x{{.FungibleTokenAddress}}
+import NonFungibleToken from 0x{{.NonFungibleTokenAddress}}
+import MetadataViews from 0x{{.MetadataViewsAddress}}
+
+/// The English Premier League NFT and metadata contract
+///
+pub contract EnglishPremierLeague: NonFungibleToken {
+
+    /// Contract events
+    ///
+    pub event ContractInitialized()
+
+    pub event Withdraw(id: UInt64, from: Address?)
+
+    pub event Deposit(id: UInt64, to: Address?)
+
+    pub event SeriesCreated(id: UInt64, name: String)
+
+    pub event SeriesClosed(id: UInt64)
+
+    pub event SetCreated(id: UInt64, name: String)
+
+    pub event SetLocked(setID: UInt64)
+
+    pub event PlayCreated(id: UInt64, metadata: {String: String}, tagIds: [UInt64])
+
+    pub event TagCreated(id: UInt64, name: String)
+
+    pub event EditionCreated(
+        id: UInt64,
+        seriesID: UInt64,
+        setID: UInt64,
+        playID: UInt64,
+        maxMintSize: UInt64?,
+        tier: String,
+    )
+
+    pub event EditionClosed(id: UInt64)
+
+    pub event MomentNFTMinted(id: UInt64, editionID: UInt64)
+
+    pub event MomentNFTBurned(id: UInt64,  editionID: UInt64)
+
+    /// Named Paths
+    ///
+    pub let CollectionStoragePath:  StoragePath
+    pub let CollectionPublicPath:   PublicPath
+    pub let AdminStoragePath:       StoragePath
+    pub let MinterPrivatePath:      PrivatePath
+
+    /// Contract variables
+    ///
+    pub var royaltyAddress: Address
+    pub var totalSupply:        UInt64
+    pub var nextSeriesID:       UInt64
+    pub var nextSetID:          UInt64
+    pub var nextTagID:          UInt64
+    pub var nextPlayID:         UInt64
+    pub var nextEditionID:      UInt64
+
+    /// Metadata Dictionaries
+    ///
+    access(self) let seriesIDByName:    {String: UInt64}
+    access(self) let seriesByID:        {UInt64: Series}
+    access(self) let setIDByName:       {String: UInt64}
+    access(self) let setByID:           {UInt64: Set}
+    access(self) let tagByID:           {UInt64: Tag}
+    access(self) let playByID:          {UInt64: Play}
+    access(self) let editionByID:       {UInt64: Edition}
+
+    /// A top-level Series with a unique ID and name
+    ///
+    pub struct Series {
+        pub let id: UInt64
+        pub let name: String
+        pub var active: Bool
+
+        init (id: UInt64, name: String) {
+            if let series = EnglishPremierLeague.seriesByID[id] {
+                self.id = series.id
+                self.name = series.name
+                self.active = series.active
+            } else {
+                self.id = id
+                self.name = name!
+                self.active = true
+            }
+        }
+
+        pub fun close() { self.active = false }
+    }
+
+    /// Series getters
+    ///
+    pub fun getSeries(id: UInt64): EnglishPremierLeague.Series {
+        return EnglishPremierLeague.seriesByID[id]!
```

Comment:
It would be better to return the optional values because we don't want to panic in getters if no value exists for a given `id`.

Reply:
Done. Thank you.